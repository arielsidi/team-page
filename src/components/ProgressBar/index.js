//@flow
import React from "react";
import styled from "styled-components";

const StyledProgressBar = styled.div`
    background: ${props => props.theme.appBgColor};
    border-radius: ${props => props.theme.borderRadius};
    width: 100%;
    height: 100%;
    div {
        background: ${props => props.theme.primaryColor};
        height: 100%;
        border-top-left-radius: ${props => props.theme.borderRadius};
        border-bottom-left-radius: ${props => props.theme.borderRadius};
    }
`;

const ProgressBar = (props: { value: number, maxValue: number }) => {
    const { value, maxValue } = props;
    const progress = maxValue ? Math.round((value * 100) / maxValue) : 0;
    const title = `${value}/${maxValue}`;
    return (
        <StyledProgressBar title={title}>
            <div style={{ width: progress + "%" }} title={title} />
        </StyledProgressBar>
    );
};

export default ProgressBar;
