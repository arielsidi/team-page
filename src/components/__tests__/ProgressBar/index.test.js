import React from "react";
import ProgressBar from "../../ProgressBar";
import renderer from "react-test-renderer";

describe("Progress bar", () => {
    it("Should render a progress bar with value 50 and maxvalue 100.", () => {
        const tree = renderer
            .create(<ProgressBar value="50" maxValue="100" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
