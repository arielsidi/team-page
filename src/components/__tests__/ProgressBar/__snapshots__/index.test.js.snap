// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress bar Should render a progress bar with value 50 and maxvalue 100. 1`] = `
<div
  className="ProgressBar__StyledProgressBar-sc-1nnf4cm-0 bAMphD"
  title="50/100"
>
  <div
    style={
      Object {
        "width": "50%",
      }
    }
    title="50/100"
  />
</div>
`;
